### PR TITLE
Support the updated parsing of event capabilities in MSC2762

### PR DESCRIPTION
**Pull request by turt2live**
Updates:
* https://github.com/matrix-org/matrix-doc/pull/2762/commits/8ad19262f68bda1e57e369a0245b618bbc9841a1
* https://github.com/matrix-org/matrix-doc/pull/2762/commits/3aa58ee029f923ecdffe8b307fd328d984881647

In slightly more detail: this introduces the escaping mechanics as well as limiting the parsing to expected event types.

### CODE REVIEW

File: src/models/WidgetEventCapability.ts [hunk 1]
```diff
@@ -56,6 +56,37 @@ export class WidgetEventCapability {
         return false;
     }
 
+    public static forStateEvent(
+        direction: EventDirection,
+        eventType: string,
+        stateKey?: string,
+    ): WidgetEventCapability {
+        // TODO: Enable support for m.* namespace once the MSC lands.
```
Comment (jryans) [inline src/models/WidgetEventCapability.ts:64]: Is there an issue tracking this?

File: src/models/WidgetEventCapability.ts [hunk 2]
```diff
@@ -96,11 +127,37 @@ export class WidgetEventCapability {
             // so we split on that. However, a # is also valid in either one of those so we
             // join accordingly.
             // Eg: `m.room.message##m.text` is "m.room.message" event with msgtype "#m.text".
+            const expectingKeyStr = eventSegment.startsWith("m.room.message#") || isState;
             let keyStr: string = null;
-            if (eventSegment.includes('#')) {
+            if (eventSegment.includes('#') && expectingKeyStr) {
+                // Dev note: regex is difficult to write, so instead the rules are manually written
```
Comment (jryans) [inline src/models/WidgetEventCapability.ts:133]: It's a bit sad we can't use an object and need to pack strings like this... 😓

**Review by jryans — APPROVED** [review]

File: src/models/WidgetEventCapability.ts (hunk 2 above)
Comment (turt2live) [reply src/models/WidgetEventCapability.ts:133]: Yea... A future MSC should fix this.

File: src/models/WidgetEventCapability.ts (hunk 1 above)
Comment (turt2live) [reply src/models/WidgetEventCapability.ts:64]: There will be :D

Comment (turt2live) [reply src/models/WidgetEventCapability.ts:64]: // https://github.com/matrix-org/matrix-widget-api/issues/22

**Pull request merged**